app/api/helpers/tasks.py

```diff
@@ -548,3 +559,212 @@ def delete_translations(self, zip_file_path):
+        order_expiry_time = get_settings()['order_expiry_time']
+        TicketHolder.query.filter(TicketHolder.order_id == None, TicketHolder.deleted_at.is_(None),
+                                  TicketHolder.created_at + datetime.timedelta(minutes=order_expiry_time)
+                                  < datetime.datetime.utcnow()).delete(synchronize_session=False)
+        db.session.commit()
+
+
+@celery.task(base=RequestContextTask, name='event.invoices.mark.due')
+def event_invoices_mark_due():
+    from app.instance import current_app as app
+    with app.app_context():
+        db.session.query(EventInvoice).filter(
+            EventInvoice.status == 'upcoming',
+            Event.id == EventInvoice.event_id,
+            Event.ends_at >= datetime.datetime.now(),
+            (EventInvoice.created_at + datetime.timedelta(days=30) <= datetime.datetime.now())
+        ).update({EventInvoice.status: 'due'}, synchronize_session=False)
+
+
+@celery.task(base=RequestContextTask, name='send.monthly.event.invoice')
+def send_monthly_event_invoice():
+    from app.instance import current_app as app
+    with app.app_context():
+        events = Event.query.filter_by(deleted_at=None, state='published').all()
+        for event in events:
+            # calculate net & gross revenues
+            user = event.owner
+            admin_info = get_settings()
+            currency = event.payment_currency
+            ticket_fee_object = db.session.query(TicketFees).filter_by(currency=currency).one()
+            ticket_fee_percentage = ticket_fee_object.service_fee
+            ticket_fee_maximum = ticket_fee_object.maximum_fee
+            orders = Order.query.filter_by(event=event).all()
+            gross_revenue = event.calc_monthly_revenue()
+            ticket_fees = event.tickets_sold * (ticket_fee_percentage / 100)
+            if ticket_fees > ticket_fee_maximum:
+                ticket_fees = ticket_fee_maximum
+            net_revenue = gross_revenue - ticket_fees
+            payment_details = {
+                'tickets_sold': event.tickets_sold,
+                'gross_revenue': gross_revenue,
+                'net_revenue': net_revenue,
+                'amount_payable': ticket_fees
+            }
+            # save invoice as pdf
+            pdf = create_save_pdf(render_template('pdf/event_invoice.html', orders=orders, user=user,
+                                  admin_info=admin_info, currency=currency, event=event,
+                                  ticket_fee_object=ticket_fee_object, payment_details=payment_details,
+                                  net_revenue=net_revenue), UPLOAD_PATHS['pdf']['event_invoice'],
+                                  dir_path='/static/uploads/pdf/event_invoices/', identifier=event.identifier)
+            # save event_invoice info to DB
+
+            event_invoice = EventInvoice(amount=net_revenue, invoice_pdf_url=pdf, event_id=event.id)
+            save_to_db(event_invoice)
```

Comment:
Why move these tasks here? Let them be in the file where they were

Reply:
@iamareebjamal - because we need to register these task in celery. keeping them in the other file will not register those task.

https://github.com/fossasia/open-event-server/blob/0ccf18cec492ce039020f9aa7a9c815436a33892/app/instance.py#L233-L238

Reply:
Need to think of some other way

Reply:
@iamareebjamal - The end goal is to keep the task of scheduled job seperate?

Reply:
Yes, to make as little changes as possible

Reply:
Let me check and get back to this, thankyou!